### ISSUE TITLE
'Iceberg column type timestamptz is not supported' when querying an Iceberg table in AWS Glue

### ISSUE THREAD

**Issue by denodo-research-labs** [prestodb/presto#22794]
When querying an Iceberg table in AWS Glue the following error is received: 'Iceberg column type timestamptz is not supported'.
This worked before Presto 0.285. It is related to #21096

## Your Environment
* Presto version used: 0.286
* Storage (HDFS/S3/GCS..): Glue with S3
* Data source and connector used: Iceberg connector
* Deployment (Cloud or On-prem): AWS Cloud

## Expected Behavior
Presto should be able to query Iceberg tables with timestamptz columns registered in AWS Glue, as it was before Presto 0.285.

## Current Behavior
 Query failed: Iceberg column type timestamptz is not supported

```
...
Caused by:  java.lang.UnsupportedOperationException: Iceberg column type timestamptz is not supported
	at com.facebook.presto.iceberg.ColumnIdentity.createColumnIdentity(ColumnIdentity.java:130)
	at com.facebook.presto.iceberg.ColumnIdentity.createColumnIdentity(ColumnIdentity.java:124)
	at com.facebook.presto.iceberg.IcebergColumnHandle.create(IcebergColumnHandle.java:184)
	at com.facebook.presto.iceberg.IcebergUtil.lambda$getColumns$3(IcebergUtil.java:265)
...
```

**Comment by tdcmeehan** [comment]
@denodo-research-labs I believe https://github.com/prestodb/presto/pull/21096 only intended to disallow adding a new column.  We can backport to previous releases once a fix is out, if you'd like to fix that please let me know.

**Comment by denodo-research-labs** [comment]
Yes, we will try to submit a fix in the next few days.

**Comment by tdcmeehan** [comment]
Fixed by #22926
